cpp/src/parquet/bloom_filter.cc

```diff
@@ -65,51 +69,109 @@ void BlockSplitBloomFilter::Init(const uint8_t* bitset, uint32_t num_bytes) {
   PARQUET_ASSIGN_OR_THROW(data_, ::arrow::AllocateBuffer(num_bytes_, pool_));
   memcpy(data_->mutable_data(), bitset, num_bytes_);
 
-  this->hasher_.reset(new MurmurHash3());
+  this->hasher_ = std::make_unique<XxHasher>();
 }
 
-BlockSplitBloomFilter BlockSplitBloomFilter::Deserialize(ArrowInputStream* input) {
-  uint32_t len, hash, algorithm;
-  int64_t bytes_available;
+static constexpr uint32_t kBloomFilterHeaderSizeGuess = 32;
+static constexpr uint32_t kMaxBloomFilterHeaderSize = 1024;
 
-  PARQUET_ASSIGN_OR_THROW(bytes_available, input->Read(sizeof(uint32_t), &len));
-  if (static_cast<uint32_t>(bytes_available) != sizeof(uint32_t)) {
-    throw ParquetException("Failed to deserialize from input stream");
+static ::arrow::Status ValidateBloomFilterHeader(
+    const format::BloomFilterHeader& header) {
+  if (!header.algorithm.__isset.BLOCK) {
+    std::stringstream ss;
+    ss << "Unsupported Bloom filter algorithm: " << header.algorithm << ".";
+    return ::arrow::Status::Invalid(ss.str());
   }
 
-  PARQUET_ASSIGN_OR_THROW(bytes_available, input->Read(sizeof(uint32_t), &hash));
-  if (static_cast<uint32_t>(bytes_available) != sizeof(uint32_t)) {
-    throw ParquetException("Failed to deserialize from input stream");
+  if (!header.hash.__isset.XXHASH) {
+    std::stringstream ss;
+    ss << "Unsupported Bloom filter hash: " << header.hash << ".";
+    return ::arrow::Status::Invalid(ss.str());
```

Comment:
You should be able to write this more easily:
```suggestion
    return ::arrow::Status::Invalid("Unsupported Bloom filter hash: ", header.hash, ".";);
```

Reply:
Yeah, this is add by myself but I forgot it, lol...